src/page-multi-facility/MultiFacilityImpactDashboard.tsx

```diff
@@ -75,23 +78,43 @@ const ScenarioTab = styled.li<{ active?: boolean }>`
 
 const MultiFacilityImpactDashboard: React.FC = () => {
   const rejectionToast = useRejectionToast();
+  const referenceFacilitiesEligible = useReferenceFacilitiesEligible();
   const showRateOfSpreadTab = useFlag(["showRateOfSpreadTab"]);
+
   const { data: localeDataSource } = useLocaleDataState();
-  const [scenario] = useScenario();
-  const scenarioId = scenario?.data?.id;
+  const [referenceDataModalOpen, setReferenceDataModalOpen] = useState(false);
+  const [scenarioState] = useScenario();
+  const scenario = scenarioState?.data;
+  const scenarioId = scenario?.id;
   const {
     state: facilitiesState,
     actions: { createOrUpdateFacility, deselectFacility },
   } = useFacilities();
-  const facilities = Object.values(facilitiesState.facilities);
+  const facilities = Object.values(facilitiesState.facilities) || [];
   const rtData = getFacilitiesRtDataById(facilitiesState.rtData, facilities);
+  const systemType = facilities[0]?.systemType;
+  const stateName = facilities[0]?.modelInputs.stateName;
+  const newlyAddedReferenceData =
```

Comment:
Oh yeah, regarding opening for baseline scenarios--I think it's currently working since it's also checking the value from line 81:
`const referenceFacilitiesEligible = useReferenceFacilitiesEligible();`

which is checking that the `enableShadowData` flag is on and that we are looking at a baseline scenario:
 
`useFlag(["enableShadowData"]) && !!scenario.data?.baseline;`

Re: seeing the modal when all facilities are already mapped, I totally forgot that I talked to @sychang about adding a default option that is `Add as new facility` so I can add that in! Good catch